cmd/buffer-worker/main.go

```diff
@@ -74,19 +73,13 @@ func run() error {
 	tel, err := telemetry.New(
 		func() (trace.TracerProvider, error) {
 			if cfg.DatadogEnabled {
-				tracerProvider := ddotel.NewTracerProvider(
-					tracer.WithLogger(telemetry.NewDDLogger(logger)),
+				return telemetry.NewDDTracerProvider(
+					logger, proc,
 					tracer.WithRuntimeMetrics(),
 					tracer.WithSamplingRules([]tracer.SamplingRule{tracer.RateRule(1.0)}),
 					tracer.WithAnalyticsRate(1.0),
 					tracer.WithDebugMode(cfg.DatadogDebug),
-				)
-				proc.OnShutdown(func() {
-					if err := tracerProvider.Shutdown(); err != nil {
-						logger.Error(err)
-					}
-				})
-				return tracerProvider, nil
```

Comment:
Tu nam chybalo `return telemetry.WrapDD(tracerProvider` ako workaround pre bug, blizsie popisane vo `wrapDD` funkcii.

Aby sa to uz nestalo, tak som spravil radsej novu funkciu `NewDDTracerProvider`, ktora to vsade (TA, BA, BW)  rovnako nasetupuje.